pkg/util/priority/priority_test.go

```diff
@@ -137,29 +143,87 @@ func TestGetPriorityFromPriorityClass(t *testing.T) {
 			builder := fake.NewClientBuilder().WithScheme(scheme).WithLists(tt.priorityClassList)
 			client := builder.Build()
 
-			name, value, err := GetPriorityFromPriorityClass(context.Background(), client, tt.priorityClassName)
-			if tt.wantErr != "" {
-				if err == nil {
-					t.Fatalf("expected an error")
-				}
-
-				if diff := cmp.Diff(tt.wantErr, err.Error()); diff != "" {
-					t.Errorf("unexpected error (-want,+got):\n%s", diff)
-				}
-				return
-			}
-
-			if err != nil {
-				t.Fatalf("unexpected error: %v", err)
+			name, source, value, err := GetPriorityFromPriorityClass(context.Background(), client, tt.priorityClassName)
+			if diff := cmp.Diff(tt.wantErr, err); diff != "" {
+				t.Errorf("unexpected error (-want,+got):\n%s", diff)
 			}
 
 			if name != tt.wantPriorityClassName {
 				t.Errorf("unexpected name: got: %s, expected: %s", name, tt.wantPriorityClassName)
 			}
 
+			if source != tt.wantPriorityClassSource {
+				t.Errorf("unexpected source: got: %s, expected: %s", source, tt.wantPriorityClassSource)
+			}
+
 			if value != tt.wantPriorityClassValue {
 				t.Errorf("unexpected value: got: %d, expected: %d", value, tt.wantPriorityClassValue)
 			}
 		})
 	}
 }
+
+func TestGetPriorityFromWorkloadPriorityClass(t *testing.T) {
+	scheme := runtime.NewScheme()
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding kueue scheme: %v", err)
+	}
+
+	tests := map[string]struct {
+		workloadPriorityClassList       *kueue.WorkloadPriorityClassList
+		workloadPriorityClassName       string
+		wantWorkloadPriorityClassName   string
+		wantWorkloadPriorityClassSource string
+		wantWorkloadPriorityClassValue  int32
+		wantErr                         error
+	}{
+		"workloadPriorityClass is specified and it exists": {
+			workloadPriorityClassList: &kueue.WorkloadPriorityClassList{
+				Items: []kueue.WorkloadPriorityClass{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "test"},
+						Value:      50,
+					},
+				},
+			},
+			workloadPriorityClassName:       "test",
+			wantWorkloadPriorityClassSource: constants.WorkloadPriorityClassSource,
+			wantWorkloadPriorityClassName:   "test",
+			wantWorkloadPriorityClassValue:  50,
+		},
+		"workloadPriorityClass is specified and it does not exist": {
+			workloadPriorityClassList: &kueue.WorkloadPriorityClassList{
+				Items: []kueue.WorkloadPriorityClass{},
+			},
+			workloadPriorityClassName: "test",
+			wantErr:                   apierrors.NewNotFound(kueue.Resource("workloadpriorityclasses"), "test"),
+		},
+	}
+
+	for desc, tt := range tests {
+		tt := tt
+		t.Run(desc, func(t *testing.T) {
+			t.Parallel()
+
+			builder := fake.NewClientBuilder().WithScheme(scheme).WithLists(tt.workloadPriorityClassList)
+			client := builder.Build()
+
+			name, source, value, err := GetPriorityFromWorkloadPriorityClass(context.Background(), client, tt.workloadPriorityClassName)
+			if diff := cmp.Diff(tt.wantErr, err); diff != "" {
```

Comment:
```suggestion
			if diff := cmp.Diff(tt.wantErr, err, cmpopts.EquateErrors()); diff != "" {
```

Reply:
When I added `cmpopts.EquateErrors`, I got an error below.
Seems like if error object's pointer is different it returns error. This is not expected.

> 
>           (*errors.StatusError)(
>         - 	&⟪0xc000448f00⟫{
>         - 		ErrStatus: "k8s.io/apimachinery/pkg/apis/meta/v1".Status{
>         - 			Status:  "Failure",
>         - 			Message: `priorityclasses.scheduling.k8s.io "test" not found`,
>         - 			Reason:  "NotFound",
>         - 			Details: &⟪0xc00003e3c0⟫"k8s.io/apimachinery/pkg/apis/meta/v1".StatusDetails{Name: "test", Group: "scheduling.k8s.io", Kind: "priorityclasses"},
>         - 			Code:    404,
>         - 		},
>         - 	},
>         + 	&⟪0xc0002f60a0⟫{
>         + 		ErrStatus: "k8s.io/apimachinery/pkg/apis/meta/v1".Status{
>         + 			Status:  "Failure",
>         + 			Message: `priorityclasses.scheduling.k8s.io "test" not found`,
>         + 			Reason:  "NotFound",
>         + 			Details: &⟪0xc00012e540⟫"k8s.io/apimachinery/pkg/apis/meta/v1".StatusDetails{Name: "test", Group: "scheduling.k8s.io", Kind: "priorityclasses"},
>         + 			Code:    404,
>         + 		},
>         + 	},
>           )

Reply:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1081/pull-kueue-test-unit-main/1702329955154661376

Reply:
I see. Ideally, wrapping an error in https://github.com/kubernetes-sigs/kueue/blob/540551609352332c8d5877a6e1a025030102facb/pkg/util/priority/priority.go#L50, then we compare the unwrapped error. However, in this case, the way might be overkill.